### PR TITLE
adds iam-units to dependency check

### PR DESCRIPTION
**Pull request by gidden**
Small hotfix building on #531 which adds a bit of documentation to `setup.py` and includes `iam-units` in dependencies which are kept up to date in the new workflow.

### CODE REVIEW

**Review by danielhuppmann — APPROVED** [review]
Thanks @gidden, lgtm!

**Comment by danielhuppmann** [comment]
Oh, seems that the explicit iam-units version actually doesn't work...?

File: .github/workflows/pytest-dependency.yml [hunk 1]
```diff
@@ -23,8 +23,9 @@ jobs:
         python-version: '3.7'
 
     - name: Install specific out-dated version of dependencies
-      run: pip install pandas==1.1.1 matplotlib==3.2.0
-      # Update the corresponding package requirements when changing dependency versions
+      # Update the corresponding package requirements when changing
+      # minimum dependency versions
+      run: pip install pandas==1.1.1 matplotlib==3.2.0 iam-units==2020.4.12
```
Comment (danielhuppmann) [inline .github/workflows/pytest-dependency.yml:28]:
```suggestion
      run: pip install pandas==1.1.1 matplotlib==3.2.0 iam-units==2020.4.21
```

File: setup.py [hunk 2]
```diff
@@ -11,6 +11,10 @@
   \/_/     \/_____/   \/_/\/_/   \/_/  \/_/
 """
 
+# NOTE TO DEVS
+# If you change a minimum version below, please explicitly set that
+# in our minimum-reqs test in the file
+# ./.github/workflows/pytest-depedency.yml
 REQUIREMENTS = [
     "argparse",
     "iam-units>=2020.4.12",
```
Comment (danielhuppmann) [inline setup.py:20]:
```suggestion
    "iam-units>=2020.4.21",
```

**Comment by danielhuppmann** [comment]
Seems that support for CO2e was added with https://github.com/IAMconsortium/units/pull/21, with the corresponding release on 2020.4.21

**Comment by gidden** [comment]
Added the commits - let's see how the pipelines work out. Appears the test is already working, though =)

**Comment by danielhuppmann** [comment]
Thanks @gidden

**Pull request merged**